cartoframes/viz/map.py

```diff
@@ -190,8 +191,47 @@ def publish(self, name, maps_api_key='default_public', context=None, password=No
             raise CartoException('The map layers are not synchronized with CARTO. '
                                  'Please, use the `sync_data` method before publishing the map')
 
+        if maps_api_key == 'default_public' and not self._publisher.is_public():
+            raise CartoException('The datasets used in your map are not public. '
+                                 'You need add new Regular API key with permissions to Maps API and the datasets. '
+                                 'You can do it from your CARTO dashboard or using the Auth API. You can get more '
+                                 'info at https://carto.com/developers/auth-api/guides/types-of-API-Keys/')
+
+
         self._publisher.set_context(context)
+        html = self._get_publication_html(name, maps_api_key)
+        self._kuviz = self._publisher.publish(html, name, password)
+        self._kuviz_message()
+
+    def sync_data(self, table_name, context=None):
+        if not self._publisher.is_sync():
+            self._publisher.sync_layers(table_name, context)
+
+    def delete_publication(self):
+        if self._kuviz:
+            self._kuviz.delete()
+            self._kuviz = None
+            print("Publication deleted")
+
+    def update_publication(self, name, password, maps_api_key='default_public', context=None):
```

Comment:
What about consolidating this into `.publish` with a new `update=True` flag?

Reply:
It could be, but I prefer to use a method only for one thing.